Sources/HTTP/HTTPServer.swift

```diff
@@ -0,0 +1,46 @@
+// This source file is part of the Swift.org Server APIs open source project
+//
+// Copyright (c) 2017 Swift Server API project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+//
+
+/// Definition of an HTTP server.
+public protocol HTTPServer : class {
+
+    /// Start the HTTP server on the given `port`, using `handler` to process incoming requests
+    func start(port: Int, handler: @escaping HTTPRequestHandler) throws
```

Comment:
Golang's server has an `Addr string` field which seems to encapsulate both the address name and port. I'm not sure whether there's a standard format for this that we could/should use?